packages/pluggableWidgets/maps-native/src/Maps.tsx

```diff
@@ -175,67 +186,62 @@ export class Maps extends Component<Props, State> {
         );
     }
 
-    private async updateCamera(animate: boolean): Promise<void> {
+    private async updateCamera(animated: boolean): Promise<void> {
         if (!this.mapViewRef.current) {
             return;
         }
 
-        if (this.props.fitToMarkers && this.props.markers.length > 1) {
-            this.mapViewRef.current.fitToElements(animate);
-            return;
-        }
-
         const camera = {
             center: await this.getCenter(),
             zoom: toZoomValue(this.props.defaultZoomLevel),
             altitude: toAltitude(this.props.defaultZoomLevel)
         };
 
-        if (animate) {
-            this.mapViewRef.current.animateCamera(camera);
+        if (this.props.fitToMarkers && this.state.markers && this.state.markers.length > 1) {
+            const coords = (this.state.markers as Marker[]).map(marker => marker.coordinate);
+            this.mapViewRef.current.fitToCoordinates(coords, { animated });
         } else {
-            this.mapViewRef.current.setCamera(camera);
+            if (animated) {
+                this.mapViewRef.current.animateCamera(camera);
+            } else {
+                this.mapViewRef.current.setCamera(camera);
+            }
         }
     }
 
     private async getCenter(): Promise<LatLng> {
+        const { fitToMarkers, centerLatitude, centerLongitude, centerAddress } = this.props;
         const center =
-            this.props.markers.length === 1 && this.props.fitToMarkers
+            (centerLatitude && centerLongitude) || centerAddress
                 ? await this.parseCoordinate(
-                      this.props.markers[0].latitude,
-                      this.props.markers[0].longitude,
-                      this.props.markers[0].address
+                      Number(centerLatitude?.value),
```

Comment:
If value is by mistake undefined it will be converted to 0, which means 0 degrees.. it will position the center in the middle of the world. Is that desired?

Reply:
You sure? Because `Number(undefined) = NaN` & `Number(null?.value) = NaN`